src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -470,10 +470,16 @@ def _validate_vm_create_storage_profile(cmd, namespace, for_scale_set=False):
             image_data_disks = [{'lun': disk.lun} for disk in image_data_disks]
 
         elif res['type'].lower() == 'galleries':
-            image_info = compute_client.gallery_images.get(resource_group_name=res['resource_group'],
-                                                           gallery_name=res['name'],
-                                                           gallery_image_name=res['child_name_1'])
-            namespace.os_type = image_info.os_type
+            try:
+                image_info = compute_client.gallery_images.get(
+                    resource_group_name=res['resource_group'],
+                    gallery_name=res['name'],
+                    gallery_image_name=res['child_name_1'])
+                namespace.os_type = image_info.os_type
+            except Exception:  # pylint: disable=broad-except
+                logger.warning('Failed to get shared gallery image definition.')
+                if not namespace.os_type:
+                    namespace.os_type = 'Linux'
```

Comment:
What if the image’s os type is Windows?

Reply:
User will see "OS type is not specified. Assume it is Linux."

Reply:
Would it cause other problems for the created vmss?